lib/host.rb

```diff
@@ -511,31 +511,24 @@ def delete(file, opts={})
         if bad_files.include? file
           raise "should not remove file named '#{file}'"
         end
-
-        r = "-r"
+        meth = :rm
       else
-        r = ""
+        meth = :remove
```

Comment:
`:rm` and `:remove` are aliases in `Fileutils`  

```ruby
  alias remove rm
  module_function :remove
```

Reply:
you should add method under `module LocalHost`. Then any local host class should pick it up by including the module. Presently you are changing the generic linux like host and it can be a remote host.

Reply:
if we strip out the Linux `~/` stuff, otherwise under LocalLinuxHost